Add SvelteKit as an option for static_site_generator setting

AndrewLester opened the pull request:
Add SvelteKit as an option for use as a static site generator.

I had set up the config parsing changes earlier today, but I added the actual SvelteKit config code more recently based on @NatoBoram's [snippet](https://github.com/actions/upload-pages-artifact/issues/9#issuecomment-1210011218). In the snippet, he provided an example for setting a `GITHUB_PAGES` environment variable, which has been discussed in [the same thread](https://github.com/actions/upload-pages-artifact/issues/9). I didn't include this just yet, as I might suggest moving it outside of the SvelteKit SSG option into a more global spot, since it may not be SvelteKit specific. This all depends on the outcome of the thread, though.

NatoBoram commented:
> an example for setting a `$GITHUB_PAGES` environment variable, which has been discussed in https://github.com/actions/upload-pages-artifact/issues/9. I didn't include this just yet, as I might suggest moving it outside of the SvelteKit SSG option into a more global spot, since it may not be SvelteKit specific

I doubt that `$VERCEL`, `$CF_PAGES` and `$NETLIFY` are SvelteKit-specific, but those are still the ones used by SvelteKit [here](https://github.com/sveltejs/kit/blob/master/packages/adapter-auto/adapters.js), so I think it's okay. After all, if something else relies on `$GITHUB_PAGES` for whatever reason, it probably also wants it to be `true` in the context where this action is used anyway

AndrewLester commented:
> This is actually something I had added in #15 (which is now merged) as well. 😊 👍🏻
> 
> Unfortunately, that means some merge conflicts here. 😅

No worries, I just rebased on those changes. I also did add the `GITHUB_PAGES` env variable export to `index.js`, as I think that's where it could live depending on our thoughts in [this thread](https://github.com/actions/upload-pages-artifact/issues/9).

NatoBoram commented:
I just re-read some comments and I noticed that @yoannchaudet [considers](https://github.com/actions/upload-pages-artifact/issues/9#issuecomment-1208771708) `$GITHUB_PAGES` and `actions/configure-pages`'s own outputs to

> pollute the environment (with Svelte specific environment variables)

Maybe we shouldn't put it in `index.js` after all? Let's put it back in `src/set-pages-path.js` and hope that someone merges it quickly so we can continue https://github.com/sveltejs/kit/pull/5845

AndrewLester commented:
I see what you mean about his comment, but in my opinion exporting such a general variable in SvelteKit only configuration is a brittle solution that breaks ownership principles.

I see two specific flaws with it. First, people may end up wanting to use the variable in other configurations since it's so general, and it has to be moved (or people may forget about it). Second, the method it would be placed in should only handle configuration setup, and giving it another task of setting the environment variable would be confusing.

I do agree that making this sveltekit only could have it get merged faster, but it's hard to tell because I don't think we've really done any convincing. I just gave you collaborator access to my repo if you want to move it, but I'd suggest moving it to another, separate, environment variable setup method similar to config setup that checks for SvelteKit.

yoannchaudet commented on src/index.js:
```diff
@@ -18,6 +18,7 @@ async function main() {
       setPagesPath({ staticSiteGenerator, generatorConfigFile, path: siteUrl.pathname })
     }
     outputPagesBaseUrl(siteUrl)
+    core.exportVariable('GITHUB_PAGES', true)
```
We had the discussion with the team and we are fine having a global environment variable exposed here.

yoannchaudet reviewed (approved):
Looking good to me. @JamesMGreene opinion :)?

JamesMGreene commented on src/index.js:
```diff
@@ -18,6 +18,7 @@ async function main() {
       setPagesPath({ staticSiteGenerator, generatorConfigFile, path: siteUrl.pathname })
     }
     outputPagesBaseUrl(siteUrl)
+    core.exportVariable('GITHUB_PAGES', true)
```
Since environment variables are all strings, let's be a little more explicit with the data type here. 👍🏻 

```suggestion
    core.exportVariable('GITHUB_PAGES', 'true')
```

NatoBoram replied:
Isn't that like saying to use a string literal instead of an enum?

JamesMGreene commented on dist/index.js:
```diff
@@ -16429,6 +16443,7 @@ async function main() {
       setPagesPath({ staticSiteGenerator, generatorConfigFile, path: siteUrl.pathname })
     }
     outputPagesBaseUrl(siteUrl)
+    core.exportVariable('GITHUB_PAGES', true)
```
Same comment but for the built distributable version. 😊 

> Since environment variables are all strings, let's be a little more explicit with the data type here. 👍🏻 

```suggestion
    core.exportVariable('GITHUB_PAGES', 'true')
```

JamesMGreene reviewed (approved):
Looks good to me, minus one nit about the data type of the environment variable export. 👍🏻

JamesMGreene commented:
This is now released in [`v1.2.0`](https://github.com/actions/configure-pages/releases/tag/v1.2.0)! 🚀 

You can use it with any of the following `uses:` statements (from most secure/specific to least):

```yaml
uses: actions/configure-pages@f71d3d08f0abfe9cbb23e527dc9f633beabf97bc
uses: actions/configure-pages@v1.2.0
uses: actions/configure-pages@v1
uses: actions/configure-pages@main
```

Thank you for the contribution! 💝
